tests/erlang_tests/test_integer_to_list.erl

```diff
@@ -56,3 +62,31 @@ compare_list([H_A | T_A], [H_B | T_B]) when H_A == H_B ->
     compare_list(T_A, T_B);
 compare_list(_A, _B) ->
     0.
+
+test_bases() ->
+    "0" = integer_to_list(?MODULE:id(0)),
+    "0" = integer_to_list(?MODULE:id(0), 10),
+    "-1" = integer_to_list(?MODULE:id(-1)),
+    "-1010" = integer_to_list(?MODULE:id(-10), 2),
+    "1010" = integer_to_list(?MODULE:id(10), 2),
+    "A" = integer_to_list(?MODULE:id(10), 16),
+    "123456789ABCDEF0" = integer_to_list(?MODULE:id(16#123456789ABCDEF0), 16),
+    "7FFFFFFFFFFFFFFF" = integer_to_list(?MODULE:id(16#7FFFFFFFFFFFFFFF), 16),
+    "-8000000000000000" = integer_to_list(?MODULE:id(-16#8000000000000000), 16),
+    "A" = integer_to_list(?MODULE:id(10), 36),
```

Comment:
Maybe having some tests for more esoteric bases might be appropriate?  I read through the loop logic in `lltoa` and  I think I understand it, but I would be more comfortable with more examples and that would also run against OTP.